module/main.c

```diff
@@ -781,10 +752,12 @@ static void handler_HidTimer(s32 data) {
                             r_edit_dirty |= R_ALL;
                         }
                         else if (mode == M_TRACK) {
-                            if (tele_patterns[edit_pattern]
-                                    .v[edit_index + offset_index] < 32766) {
-                                tele_patterns[edit_pattern]
-                                    .v[edit_index + offset_index]++;
+                            int16_t v = tele_get_pattern_val(
+                                edit_pattern, edit_index + offset_index);
+                            if (v < 32766) {
+                                tele_set_pattern_val(edit_pattern,
+                                                     edit_index + offset_index,
+                                                     v++);
```

Comment:
Good idea. One of the things I've been (slowly) trying to do is keep like minded code together. It definitely helps spot patterns and code reductions.